driver-core/src/test/unit/com/mongodb/internal/operation/ClientBulkWriteOperationTest.java

```diff
@@ -0,0 +1,217 @@
+                new Document(),
+                clientReplaceOneOptions().upsert(true)
+        ));
+        ClientBulkWriteOperation op = new ClientBulkWriteOperation(
+                clientNamespacedReplaceOneModels,
+                options,
+                WriteConcern.ACKNOWLEDGED,
+                false,
+                getDefaultCodecRegistry());
+        //when
+        ClientBulkWriteResult result = op.execute(binding, ClusterFixture.OPERATION_CONTEXT);
+
+        //then
+        assertEquals(0, result.getInsertedCount());
+        assertEquals(1, result.getUpsertedCount());
+        assertEquals(0, result.getMatchedCount());
+        assertEquals(0, result.getModifiedCount());
+        assertEquals(0, result.getDeletedCount());
+        assertFalse(result.getVerboseResults().isPresent());
+    }
+
+    /**
+     * This test exists due to SERVER-113026 bug.
+     */
+    //TODO-JAVA-6005
+    @Test
+    void shouldUseDefaultNumberOfModifiedDocumentsWhenMissingInCursor() {
+        //given
+        mockCommandExecutionResult("{"
+                + "   cursor: {"
+                + "    id: NumberLong(0),"
+                + "    firstBatch: [ {"
+                + "        'ok': 1.0,"
+                + "        'idx': 0,"
+                + "        'n': 1,"
+                //nModified field is missing here
+                + "        'upserted': {"
+                + "          '_id': 1"
+                + "        }"
+                + "      }],"
+                + "    ns: 'admin.$cmd.bulkWrite'"
+                + "  },"
+                + "  nErrors: 0,"
+                + "  nInserted: 1,"
+                + "  nMatched: 0,"
+                + "  nModified: 0,"
+                + "  nUpserted: 1,"
+                + "  nDeleted: 0,"
+                + "  ok: 1"
+                + "}");
+        ClientBulkWriteOptions options = ClientBulkWriteOptions.clientBulkWriteOptions()
+                .ordered(false).verboseResults(true);
+        List<ClientNamespacedReplaceOneModel> clientNamespacedReplaceOneModels = singletonList(ClientNamespacedWriteModel.replaceOne(
+                NAMESPACE,
+                Filters.empty(),
+                new Document(),
+                clientReplaceOneOptions().upsert(true)
+        ));
+        ClientBulkWriteOperation op = new ClientBulkWriteOperation(
+                clientNamespacedReplaceOneModels,
+                options,
+                WriteConcern.ACKNOWLEDGED,
+                false,
+                getDefaultCodecRegistry());
+        //when
+        ClientBulkWriteResult result = op.execute(binding, ClusterFixture.OPERATION_CONTEXT);
+
+        //then
+        assertEquals(1, result.getInsertedCount());
+        assertEquals(1, result.getUpsertedCount());
+        assertEquals(0, result.getMatchedCount());
+        assertEquals(0, result.getModifiedCount());
+        assertEquals(0, result.getDeletedCount());
+        assertTrue(result.getVerboseResults().isPresent());
```

Comment:
`nModified` does not affect any of these parts of the result: this is based on the semantics of `nModified` (`nModified` is for summary results), and our knowledge that the result is going to be verbose because of the `verboseResults(true)` option. The only thing we are to assert here is that `op.execute` does not throw.